validatedeps.js

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env node
+/**
+ * Created by cmeyers on 9/20/16.
+ */
+const fs = require('fs');
+const start = new Date().getTime();
+
+const args = process.argv.slice(2);
+
+var projectPath = '';
+
+if (args.length === 1) {
+    const pathParts = args[0].split('=');
+
+    if (pathParts.length !== 2) {
+        console.error('you must specify a project path in the form -p=path/to/project');
+        process.exit(1);
+    }
+
+    projectPath = pathParts.slice(-1);
+}
+
+const workingPath = process.cwd();
+const resolvedPath = buildPath(`${workingPath}/${projectPath}`);
+console.log(`validating dependencies in ${resolvedPath}`);
+const packageJsonPath = buildPath(`${resolvedPath}/package.json`);
+const shrinkwrapJsonPath = buildPath(`${resolvedPath}/npm-shrinkwrap.json`);
+
+// match x, tilde, gt, lt, star, carat or whitespace in version
+const IMPRECISE_VERSION_CHARS_PATTERN = /[x~><*|\^\s]+/;
```

Comment:
Why don't we make sure we have `^\d+\.\d+\.\d+$` instead of trying to find imprecise ones? It looks like this wouldn't flag a version of "latest" or "3" as bad?